src/main/java/core/basesyntax/strategy/impl/PurchaseOperation.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.HashMap;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(HashMap<String, Integer> storage, FruitTransaction transaction) {
+        String fruitName = transaction.getFruit();
+        int oldQuantity = storage.get(fruitName);
```

Comment:
There's a potential `NullPointerException` if `storage.get(fruitName)` returns null because the fruitName is not present in the storage. Consider checking if the fruit is present in the storage before attempting to retrieve its quantity.